### DISCUSSION TITLE
第3节课笔记(15班) 

### DISCUSSION THREAD

**Discussion by ironartisan** [InternLM/Tutorial#38] [笔记]
请在下方提交笔记链接

**Comment by Mrguanglei** [comment]
飞书：https://gzr5xfkph4.feishu.cn/docx/W6ueduEBUoFUQ6xnC7ncfMBbnxh?from=from_copylink

**Comment by lizhunkg** [comment]
第三课笔记作业：https://note.youdao.com/s/3gVzjwmx

**Comment by szhowardhuang** [comment]
按照LangChain教程，
1） 安装环境，过程顺利，huggingface 下载 sentence transformer，要用镜像站点。model可以存在之前创建的/root/model目录里面。

![294798226-f9e8a58e-207a-4592-9b6d-685470065d40](https://github.com/InternLM/tutorial/assets/1407300/a8d64461-c5fc-4bc1-a020-7da210148927)

2） 代码下载， 可下载到code目录，不过之后脚本或者代码要注意目录
![294799022-45d4572d-a19c-44b6-96f3-9ce5004a5c73](https://github.com/InternLM/tutorial/assets/1407300/ec05639e-6e0f-452a-9ac3-9f659e26d568)

3）数据收集， 建立data目录 /root/data ， 把数据集放到此目录. 把脚本拷贝到data目录， cp -r /root/code/tutorial/langchain/demo/ ./
![294799324-3e638c5a-379b-4196-a3fc-58d221281487](https://github.com/InternLM/tutorial/assets/1407300/53a56126-abc1-4eb2-8c19-ae4325e25c86)

4）构建向量数据库， 执行create_db.py即可。注意调整model路径， 数据库文件会存到当前python执行路径
5）部署web demo，由于我用VSC remote，懒得改隧道端口。所以更改一下教程里面的server port。

![294847219-cbc63a8c-94b9-4664-b655-f511cdc0b7d0](https://github.com/InternLM/tutorial/assets/1407300/097ef36a-ad92-42df-9b41-2d325fafe190)


执行刚开始要等一下
![294847274-2131bca1-c9ee-4578-a1bb-262aa77f1927](https://github.com/InternLM/tutorial/assets/1407300/af8f9e30-2fff-4beb-ba86-06c1e3825c99)


效果如下， 类似的问题得到不同的结果，看起来要用中文切片模型。
![294847836-3f3d84ae-9c91-4298-af30-ae490f9a65c4](https://github.com/InternLM/tutorial/assets/1407300/af6d8897-9ae8-4219-bc2b-4bbbe768caf4)


为了确认是否切片模型造成乱码，下载bge-large-zh
import os
os.environ['HF_ENDPOINT'] = 'https://hf-mirror.com/'
os.system('huggingface-cli download --resume-download BAAI/bge-large-zh --local-dir /root/model/bge-large-zh')

修改构建向量数据库模型和查询向量数据库模型，记得删除之前建立的数据库。
embeddings = HuggingFaceEmbeddings(model_name="/root/model/bge-large-zh")

效果如下
![294854341-f309f387-7f57-4392-8f5f-189cd9ba5c52](https://github.com/InternLM/tutorial/assets/1407300/7bb775ba-f1fd-4a91-aa3d-3f87db3f0c90)

**Comment by scchy** [comment]
第三课笔记:
https://github.com/scchy/My_Learn/blob/master/LLM_prime/3_InternLM_langchain_doc.md

**Reply by scchy** [reply]
add openxlab app deployed picture

**Comment by woodx9** [comment]
请参照笔记部分[飞书链接](https://gxloh9mfvxh.feishu.cn/wiki/LofYwy4UsiMHO1kJOpUcR6ofnOc?from=from_copylink)

**Comment by dqyydq** [comment]
https://blog.csdn.net/qq_52410386/article/details/135498390?spm=1001.2014.3001.5501

**Comment by ziptoam** [comment]
第三节课笔记
https://note.youdao.com/s/QbcCNsph

**Comment by yabgdsajghsj** [comment]
https://note.youdao.com/s/cGGUnKdS

**Comment by IyGnahz** [comment]
笔记+作业 https://g6xuom3zbh.feishu.cn/docx/EkTidZDgaocodUxqgX0cimoKnZe

**Comment by sokolo05** [comment]
第三节课笔记https://github.com/sokolo05/Scholar_PuYu/blob/main/02.%E8%AF%BE%E7%A8%8B%E7%AC%94%E8%AE%B0/%E7%AC%AC3%E8%8A%82_%E7%AC%94%E8%AE%B0.md

**Comment by 520Enterprise** [comment]
https://www.cnblogs.com/520Enterprise/p/17945399

**Comment by Visytudz** [comment]
[3 Knowledge Base Construction HW.pdf](https://github.com/InternLM/tutorial/files/13887902/3.Knowledge.Base.Construction.HW.pdf)

**Comment by 6forwater29** [comment]
没有写笔记，在作业里面

**Comment by bblgbr** [comment]
笔记链接：
https://github.com/bblgbr/play_llm/blob/main/InterLM_Class/03%20LangChain%E6%90%AD%E5%BB%BA%E7%9F%A5%E8%AF%86%E5%BA%93.md

**Comment by Gaia-yue** [comment]
https://github.com/Gaia-yue/---/blob/main/%E4%B8%80%E6%9C%88/3_langchain-note.md

**Comment by huiruije** [comment]
作业＋笔记https://blog.csdn.net/m0_59356572/article/details/135571447?spm=1001.2014.3001.5501

**Comment by seienn** [comment]
https://gvxxiiwsufy.feishu.cn/wiki/NCClweMjkiAwxwkXXcOcqAV2nXe?from=from_copylink

**Comment by NANDSS1** [comment]
https://www.yuque.com/nandss/gyyg9p/bika0mo6cyno0bg5?singleDoc# 《lesson3 基于 InternLM 和 LangChain 搭建知识库 笔记》

**Comment by g29times** [comment]
https://novagen.notion.site/3-1e555eae5529411aaf2983a7051f791d?pvs=4
